### PR TITLE
update test.yml action to run tests with net5.0

### PR DESCRIPTION
**Pull request by valentin-p**
* update test.yml action to run tests with net5.0
* This PR should be merged to main before #887 

I have tested it on my forked repository together with #887: https://github.com/valentin-p/fsharp/runs/1802285485

### CODE REVIEW

**Review by jrr — APPROVED** [review]

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -4,12 +4,12 @@ on: [push, pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@f90c7b395dac7c5a277c1a6d93d5057c1cddb74e # v2
+      - uses: actions/checkout@v2.3.4
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:9]: Could you revert this change? We have a security policy where we fix GitHub actions to SHAs instead of tags.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -4,12 +4,12 @@ on: [push, pull_request]
 
 jobs:
   test:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
```
Comment (ErikSchierboom) [inline .github/workflows/test.yml:7]: Could you revert this change? It is usually better to use a fixed version than just silently being upgraded. I'm also fine with updating this to `ubuntu-20.04`.

**Review by ErikSchierboom — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (valentin-p) [reply .github/workflows/test.yml]: reverted

File: .github/workflows/test.yml (hunk 2 above)
Comment (valentin-p) [reply .github/workflows/test.yml]: using 20.04

**Review by ErikSchierboom — APPROVED** [review]
Great!
